### PR TITLE
Improved typing of Galaxy model store related stuff.

### PR DESCRIPTION
**Pull request by jmchilton**
Cleaning up app handling with model stores to make the deferred data branch a little cleaner.

## How to test the changes?
(Select all options that apply)
- [x] This is a refactoring of components with existing test coverage.

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

**Review by bgruening — APPROVED** [review]
The failing test is unrelated afaik.

File: lib/galaxy/model/store/__init__.py [hunk 1]
```diff
@@ -1222,20 +1256,22 @@ def __init__(
...
-        self.included_libraries = []
-        self.included_library_folders = []
-        self.collection_datasets = {}
-        self.collections_attrs = []
-        self.dataset_id_to_path = {}
+        self.included_datasets: Dict[model.DatasetInstance, Tuple[model.DatasetInstance, bool]] = {}
+        self.included_collections: List[Union[model.DatasetCollection, model.HistoryDatasetCollectionAssociation]] = []
+        self.included_libraries: List[model.Library] = []
+        self.included_library_folders: List[model.LibraryFolder] = []
+        self.collection_datasets: Set[int] = set()
```
Comment (mvdbeek) [inline lib/galaxy/model/store/__init__.py:1263]: Should we do
```suggestion
        self.collection_datasets: Dict[model.DatasetInstance, Tuple[model.DatasetInstance, bool]] = {}
```
? I guess right now we only expect HDAs, but if we ever allow other things as collection datasets we might run into odd bugs ?

**Review by mvdbeek — APPROVED** [review]

File: lib/galaxy/model/store/__init__.py (hunk 1 above)
Comment (davelopez) [reply lib/galaxy/model/store/__init__.py:1263]: Python illiterate question: do we have to redefine `__hash__()` and `__eq__()` in `model.DatasetInstance` to safely use the class as dict key? or it 'just works'?

Comment (jmchilton) [reply lib/galaxy/model/store/__init__.py:1263]: @mvdbeek Well right now we only use this to do one thing (a dict we place True into for some integer IDs - which wold be better represented as a set). If we want to use it differently in the future like you suggest we can just change it then right? It is only used in a few places and we have excellent test coverage.

Comment (mvdbeek) [reply lib/galaxy/model/store/__init__.py:1263]: I don't think tests would catch a mistake like that (identical primary keys for different items is what I'm worried about), but fair enough, we'll probably remember to do that.

**Pull request merged**